

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  <meta name="generator" content="Docutils 0.18.1: http://docutils.sourceforge.net/" />

  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Terminology and Information Model &mdash; GKS-Transcript  documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script data-url_root="./" id="documentation_options" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/_sphinx_javascript_frameworks_compat.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/theme_overrides.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="References" href="References.html" />
    <link rel="prev" title="Introduction" href="introduction.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search" >
          

          
            <a href="index.html" class="icon icon-home"> GKS-Transcript
          

          
          </a>

          
            
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <ul class="current">
<li class="toctree-l1"><a class="reference internal" href="introduction.html">Introduction</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="#">Terminology and Information Model</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#core-classes">Core Classes</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#sequence">Sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="#locatedfeature">LocatedFeature</a></li>
<li class="toctree-l3"><a class="reference internal" href="#sequencefeature">SequenceFeature</a></li>
<li class="toctree-l3"><a class="reference internal" href="#sequencecontext">SequenceContext</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="#annotated-sequence">Annotated Sequence</a><ul>
<li class="toctree-l3"><a class="reference internal" href="#id6">Annotated Sequence</a></li>
<li class="toctree-l3"><a class="reference internal" href="#transcript">Transcript</a></li>
</ul>
</li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="References.html">References</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">GKS-Transcript</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
      <li>Terminology and Information Model</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            <a href="_sources/terms_and_model.rst.txt" rel="nofollow"> View page source</a>
          
        
      </li>
    
  </ul>

  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <section id="terminology-and-information-model">
<h1>Terminology and Information Model<a class="headerlink" href="#terminology-and-information-model" title="Permalink to this heading">¶</a></h1>
<p>This is a DRAFT information model that was developed as part of the GA4GH GKS SA subgroup’s exercise to model sequence features and transcripts. It is a work in progress.</p>
<p>This information model was derived from the <a class="reference external" href="https://docs.google.com/presentation/d/1WdDSpr3XqSt9TXp_iyr2FiCBM4BANbL6JM24H1I6UjE/edit#slide=id.p">draft conceptual and logical models</a> that were developed as foundational work for this effort. Please reference those models in addition to this documentation (which is incomplete).</p>
<p>Note: Some elements are sourced from the <a class="reference external" href="https://vrs.ga4gh.org/en/latest/index.html">Variation Representation Specification (VRS)</a> and others from the <a class="reference external" href="https://build.fhir.org/branches/cg-im-molseq-work_in_progress/molecularsequence.html">HL7 Clinical Genomics FHIR Molecular Sequence Resource</a>. This draft model was aligned to those specifications as much as possible while still achieving the goals of the SA modeling exercise.</p>
<figure class="align-default" id="id8">
<span id="schema-temp"></span><img alt="_images/Generalized_SA_Model.png" src="_images/Generalized_SA_Model.png" />
<figcaption>
<p><span class="caption-text">Draft SA Model</span><a class="headerlink" href="#id8" title="Permalink to this image">¶</a></p>
<div class="legend">
<blockquote>
<div><p><strong>Legend</strong> DRAFT model of Annotated Sequence, Transcript, and building blocks (including Sequence and SequenceFeature). See the <a class="reference external" href="https://docs.google.com/presentation/d/1WdDSpr3XqSt9TXp_iyr2FiCBM4BANbL6JM24H1I6UjE/edit#slide=id.p">conceptual and logical models</a> for rationale, design choices, and examples.</p>
</div></blockquote>
</div>
</figcaption>
</figure>
<section id="core-classes">
<h2>Core Classes<a class="headerlink" href="#core-classes" title="Permalink to this heading">¶</a></h2>
<section id="sequence">
<span id="id1"></span><h3>Sequence<a class="headerlink" href="#sequence" title="Permalink to this heading">¶</a></h3>
<p>The definition of Sequence is conceptually identical to <a class="reference external" href="https://vrs.ga4gh.org/en/stable/terms_and_model.html#sequence">VRS Sequence</a>, but the information model is more detailed.</p>
<p>A <em>sequence</em> is a contiguous, linear polymer of nucleic acid or amino acid Residues. Sequences are the prevalent representation of these polymers, particularly in the domain of variant representation.</p>
<p><strong>Computational Definition</strong></p>
<p>A character string of Residues that represents a biological sequence using the conventional sequence order (5’-to-3’ for nucleic acid sequences, and amino-to-carboxyl for amino acid sequences). IUPAC ambiguity codes are permitted in Sequences.</p>
<p><strong>Information Model</strong></p>
<p>The Sequence class represents the concept of a particular sequence rather than its instantiation in a particular database or serialization. The Sequence class provides convenience attributes to capture human-readable names and identifiers that are associated with a sequence, but the <em>representation</em> of the sequence itself (the linear string of residues) is captured using the SequenceRepresentation abstract data type (defined by the <a class="reference external" href="https://build.fhir.org/branches/cg-im-molseq-work_in_progress/molecularsequence.html">FHIR Molecular Sequence Resource</a>, which are similar to the <a class="reference external" href="https://vrs.ga4gh.org/en/stable/terms_and_model.html#sequence-expression">VRS SequenceExpression classes</a>).</p>
<table class="clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>type</p></td>
<td><p><a class="reference external" href="https://build.fhir.org/datatypes.html#CodeableConcept">CodeableConcept (FHIR)</a></p></td>
<td><p>[1..1]</p></td>
<td><p>The type of sequence (e.g., DNA, RNA, protein). The value of this attribute should represent a term in an ontology (e.g., Sequence Ontology).</p></td>
</tr>
<tr class="row-odd"><td><p>identifier</p></td>
<td><p><a class="reference external" href="https://build.fhir.org/datatypes.html#Identifier">Identifier (FHIR)</a></p></td>
<td><p>[0..*]</p></td>
<td><p>Identifier(s) for the sequence. Identifiers are used for cross-referencing sequence concepts only, not for capturing the sequence itself (which is done with the <em>representation</em> attribute). Note the complex datatype includes attributes to capture both the identifer and the system or namespace that assigned the identifier. Identifiers MUST be unique within a system or namespace.</p></td>
</tr>
<tr class="row-even"><td><p>name</p></td>
<td><p>string</p></td>
<td><p>[0..*]</p></td>
<td><p>Name(s) for the sequence. Names are intended to be for human-readable purposes only and are not guaranteed to uniquely specify a sequence.</p></td>
</tr>
<tr class="row-odd"><td><p>representation</p></td>
<td><p>SequenceRepresentation</p></td>
<td><p>[0..*]</p></td>
<td><p>Representation(s) of the sequence. All representations for a given sequence MUST resolve to the same literal string.</p></td>
</tr>
<tr class="row-even"><td><p>features</p></td>
<td><p><a class="reference internal" href="#locatedfeature"><span class="std std-ref">LocatedFeature</span></a></p></td>
<td><p>[0..*]</p></td>
<td><p>A list of annotated features on the sequence. This attribute should be used only by Sequence-based structures.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="locatedfeature">
<span id="id2"></span><h3>LocatedFeature<a class="headerlink" href="#locatedfeature" title="Permalink to this heading">¶</a></h3>
<p>A <em>located feature</em> is a mapping of a sequence feature to a location on a sequence.</p>
<p><strong>Computational Definition</strong></p>
<p>TBD</p>
<p><strong>Information Model</strong></p>
<p>A mapping between a given sequence feature and its location on a given sequence.</p>
<table class="clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>feature</p></td>
<td><p><a class="reference internal" href="#sequencefeature"><span class="std std-ref">SequenceFeature</span></a></p></td>
<td><p>[1..1]</p></td>
<td><p>The feature at the given location</p></td>
</tr>
<tr class="row-odd"><td><p>location</p></td>
<td><p><a class="reference external" href="https://vrs.ga4gh.org/en/stable/terms_and_model.html#locations-and-intervals">Location (VRS)</a></p></td>
<td><p>[1..1]</p></td>
<td><p>The location of the feature</p></td>
</tr>
<tr class="row-even"><td><p>strand</p></td>
<td><p><a class="reference external" href="https://build.fhir.org/datatypes.html#CodeableConcept">CodeableConcept (FHIR)</a></p></td>
<td><p>[0..1]</p></td>
<td><p>An indicator specifying whether the feature is on the forward or reverse strand of a double-stranded sequence. If not set, the feature is assumed to be on the forward strand (by convention). If the sequence is single-stranded and this attribute is set, it must not be set to “reverse”.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="sequencefeature">
<span id="id3"></span><h3>SequenceFeature<a class="headerlink" href="#sequencefeature" title="Permalink to this heading">¶</a></h3>
<p>A <em>sequence feature</em> is a structural or functional feature that can be annotated on a Sequence.</p>
<p><strong>Computational Definition</strong></p>
<p>TBD</p>
<p><strong>Information Model</strong></p>
<p>A structural or functional feature that can be annotated on a Sequence at a defined location (interval). A given feature is not unique to a single Sequence and can be mapped to different locations on different Sequences through <em>contextualization</em>.</p>
<dl class="simple">
<dt>Examples of Sequence Features include:</dt><dd><ul class="simple">
<li><p>Gene locus</p></li>
<li><p>Exon</p></li>
<li><p>Intron</p></li>
<li><p>5’ or 3’ UTR</p></li>
<li><p>Transcription start site</p></li>
<li><p>Translation start or stop sites</p></li>
<li><p>Coding sequence (CDS)</p></li>
<li><p>Codon</p></li>
<li><p>Transcription factor binding site</p></li>
<li><p>Post-translational modification site</p></li>
<li><p>Splice donor/acceptor site</p></li>
<li><p>PolyA site</p></li>
</ul>
</dd>
</dl>
<p>Sequence Features are first-class entities that can stand on their own as independent entries in knowledge bases and be used as the subject of VA statements.</p>
<table class="clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>identifier</p></td>
<td><p><a class="reference external" href="https://build.fhir.org/datatypes.html#Identifier">Identifier (FHIR)</a></p></td>
<td><p>[0..*]</p></td>
<td><p>Identifier(s) for the sequence feature. Note the complex datatype includes attributes to capture both the identifer and the system or namespace that assigned the identifier. Identifiers MUST be unique within a system or namespace.</p></td>
</tr>
<tr class="row-odd"><td><p>name</p></td>
<td><p>string</p></td>
<td><p>[0..*]</p></td>
<td><p>Name(s) for the sequence feature. Names are intended to be for human-readable purposes only and are not guaranteed to uniquely specify a sequence feature.</p></td>
</tr>
<tr class="row-even"><td><p>type</p></td>
<td><p><a class="reference external" href="https://build.fhir.org/datatypes.html#CodeableConcept">CodeableConcept (FHIR)</a></p></td>
<td><p>[1..1]</p></td>
<td><p>The type of feature. The value of this attribute should refer to a term in an ontology (e.g., Sequence Ontology).</p></td>
</tr>
<tr class="row-odd"><td><p>contexts</p></td>
<td><p><a class="reference internal" href="#sequencecontext"><span class="std std-ref">SequenceContext</span></a></p></td>
<td><p>[0..*]</p></td>
<td><p>A list of sequence contexts for the feature. This attribute should be used only by Feature-based structures.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="sequencecontext">
<span id="id4"></span><h3>SequenceContext<a class="headerlink" href="#sequencecontext" title="Permalink to this heading">¶</a></h3>
<p>A <em>sequence context</em> is the definition of a location on a sequence.</p>
<p><strong>Computational Definition</strong></p>
<p>TBD</p>
<p><strong>Information Model</strong></p>
<p>The definition of a location on a given sequence. This mapping provides sequence context (<em>contextualization</em>) for a parent entity (e.g., a feature).</p>
<table class="clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>sequence</p></td>
<td><p><a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a></p></td>
<td><p>[1..1]</p></td>
<td><p>The Sequence that provides the context for the mapped location.</p></td>
</tr>
<tr class="row-odd"><td><p>location</p></td>
<td><p><a class="reference external" href="https://vrs.ga4gh.org/en/stable/terms_and_model.html#locations-and-intervals">Location (VRS)</a></p></td>
<td><p>[1..1]</p></td>
<td><p>A location that defines a region (interval) on the given Sequence.</p></td>
</tr>
<tr class="row-even"><td><p>strand</p></td>
<td><p><a class="reference external" href="https://build.fhir.org/datatypes.html#CodeableConcept">CodeableConcept (FHIR)</a></p></td>
<td><p>[0..1]</p></td>
<td><p>An indicator specifying whether the feature is on the forward or reverse strand of a double-stranded sequence. If not set, the feature is assumed to be on the forward strand (by convention). If the sequence is single-stranded and this attribute is set, it must not be set to “reverse”.</p></td>
</tr>
</tbody>
</table>
</section>
</section>
<section id="annotated-sequence">
<span id="id5"></span><h2>Annotated Sequence<a class="headerlink" href="#annotated-sequence" title="Permalink to this heading">¶</a></h2>
<p>An <em>annotated sequence</em> is a sequence that contains annotations (e.g., features).</p>
<section id="id6">
<h3>Annotated Sequence<a class="headerlink" href="#id6" title="Permalink to this heading">¶</a></h3>
<p><strong>Computational Definition</strong></p>
<p>TBD</p>
<p><strong>Information Model</strong></p>
<p>Conceptually, an <em>annotated sequence</em> is a sequence with associated annotations (e.g., features). Structurally, the class is a container for a Sequence and located features (and because of this an Annotated Sequence is a complex object that is neither a Sequence nor a Feature). This class may serve as a generalized parent class that can be specialized to support complex types of annotated sequences (e.g., transcripts).</p>
<table class="clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>sequence</p></td>
<td><p><a class="reference internal" href="#sequence"><span class="std std-ref">Sequence</span></a></p></td>
<td><p>[1..1]</p></td>
<td><p>The Sequence that is annotated with features.</p></td>
</tr>
<tr class="row-odd"><td><p>features</p></td>
<td><p><a class="reference internal" href="#locatedfeature"><span class="std std-ref">LocatedFeature</span></a></p></td>
<td><p>[0..*]</p></td>
<td><p>A list of annotated features on the Sequence.</p></td>
</tr>
</tbody>
</table>
</section>
<section id="transcript">
<span id="id7"></span><h3>Transcript<a class="headerlink" href="#transcript" title="Permalink to this heading">¶</a></h3>
<p>A <em>transcript</em> is a type of annotated sequence.</p>
<p><strong>Computational Definition</strong></p>
<p>TBD</p>
<p><strong>Information Model</strong></p>
<p>A <em>transcript</em> is a type of annotated sequence. It represents a single-stranded RNA sequence that may contain structural features (e.g., exons) and functional features (e.g., coding region). Because the concept of <em>transcript</em> is so widely used, and because the concepts of sequence, exon, and gene are so closely intertwined with it, an explicit model for Transcript was developed.</p>
<table class="clean-wrap docutils align-left">
<thead>
<tr class="row-odd"><th class="head"><p>Field</p></th>
<th class="head"><p>Type</p></th>
<th class="head"><p>Limits</p></th>
<th class="head"><p>Description</p></th>
</tr>
</thead>
<tbody>
<tr class="row-even"><td><p>exons</p></td>
<td><p><a class="reference internal" href="#locatedfeature"><span class="std std-ref">LocatedFeature</span></a></p></td>
<td><p>[0..*]</p></td>
<td><p>A list of sequence features representing the exons of the transcript.</p></td>
</tr>
<tr class="row-odd"><td><p>codingRegion</p></td>
<td><p><a class="reference internal" href="#locatedfeature"><span class="std std-ref">LocatedFeature</span></a></p></td>
<td><p>[0..*]</p></td>
<td><p>A list of sequence features representing the coding region of the transcript.</p></td>
</tr>
<tr class="row-even"><td><p>associatedGenes</p></td>
<td><p>identifier</p></td>
<td><p>[0..*]</p></td>
<td><p>A list of gene(s) associated with the transcript. Note: This structure provides support for fusions, prokaryotic operons, etc, but it does not specify which region(s) or feature(s) within the transcript are associated with a particular gene.</p></td>
</tr>
</tbody>
</table>
</section>
</section>
</section>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="References.html" class="btn btn-neutral float-right" title="References" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="introduction.html" class="btn btn-neutral float-left" title="Introduction" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 2023, GKS-Transcript

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>